src/useCustomCompareMemo/useCustomCompareMemo.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo, useRef } from 'react';
+
+import type { DependencyList } from 'react';
+
+type DepsAreEqual<TDeps extends DependencyList> = (savedDeps: TDeps, deps: TDeps) => boolean;
```

Comment:
Why do you use prefixes, btw? What does `T` stands for? It is considered bad practice in TS community nowadays, and described in contribution guide

Reply:
In my understanding "T" prefix mean that type is generic, but i respect rules of code. Give me a second, I fix it.